### PR TITLE
Guard tests with REQUIRES:aarch64-registered-target

### PR DESCRIPTION
**Pull request by rastogishubham**
(cherry picked from commit 579ccc3a7a65aa72787ff703144d5f804fee7005)

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

File: llvm/test/DebugInfo/ARM/swift-async.ll [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ; RUN: llc -O0 -filetype=obj < %s | llvm-dwarfdump --name n - | FileCheck %s
 ;
+; REQUIRES: arm64
```
Comment (dsandersllvm) [inline llvm/test/DebugInfo/ARM/swift-async.ll]: This is correct for the triple used in this test but I notice it's an arm64 (AArch64) triple in a ARM subdirectory which seems wrong. Should this test be using an ARM triple or be moved to the AArch64 subdir?

File: llvm/test/DebugInfo/CAS/AArch64/debug_line_and_info_dedupe.test [hunk 2]
```diff
@@ -5,6 +5,8 @@ RUN: llc -O0 -cas-friendly-debug-info --filetype=obj --cas-backend --cas=%t/cas
 RUN: llc -O0 -cas-friendly-debug-info --filetype=obj --cas-backend --cas=%t/cas --mccas-casid %t/b.ll -o %t/b.id
 RUN: llvm-cas-dump --die-refs --cas=%t/cas --casid-file %t/a.id %t/b.id | FileCheck %s
 
+; REQUIRES: arm64
```
Comment (dsandersllvm) [inline llvm/test/DebugInfo/CAS/AArch64/debug_line_and_info_dedupe.test]: One other thing to mention is that where target specific tests are in a subdirectory like ARM/ or AArch64/ you can put the requirement in lit.local.cfg to apply it to all tests in that directory (see CodeGen/${TARGET}/lit.local.cfg for an example)

File: llvm/test/tools/llvm-cas-dump/enums_split.ll [hunk 3]
```diff
@@ -25,8 +27,8 @@ target triple = "arm64-apple-macosx13.0.0"
 
 ; enum class ColorEnum {
 ;  Blue = 0,
-;  Red = 1
```
Comment (dsandersllvm) [inline llvm/test/tools/llvm-cas-dump/enums_split.ll]: Did you mean to change this bit?

**Review by dsandersllvm — APPROVED** [review]
Thanks for fixing this. LGTM without the change that moved `Red` outside the enum. I also had a couple notes

**Comment by cachemeifyoucan** [comment]
The correct spelling for the requirement is `REQUIRES: aarch64-registered-target`, not `arm64`.

Just like @dsandersllvm says, if the test is put in a sub directory that has the name of the target, usually you just put the requirement in the `lit.local.cfg` in that directory. You just need to follow the file example like: https://github.com/apple/llvm-project/blob/next/llvm/test/CodeGen/AArch64/lit.local.cfg so you don't need to repeat the require line in each test.

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

**Pull request merged**